### DISCUSSION TITLE
Uno ChromeCast blog post - Permissions on Android?

### DISCUSSION THREAD

**Discussion by ransagy** [unoplatform/uno#9553] [Q&A]
I've been trying to follow the blogpost about implementing a Chromecast Sender app in Uno Platform (and the demo at https://github.com/susairajs/UnoChromeCast ) and have a working piece as long as i use the WinUI head - in both Debug and Release builds.
For Android however, I cannot get any working version that finds any receiver chromecast devices. Did anyone come across something like that?

My immediate suspicions are either permissions or trimming, but i haven't been able to make anything tick by changing either.
I have added the INTERNET, ACCESS_NETWORK_STATE and ACCESS_WIFI_STATE permissions. Am i missing something else for the chromecast's DIAL protocol? something specific for private networks or..?

I'm not 100% sure if its specific to Uno or not just yet, either.

**Comment by johnsmith3321** [comment]
It's possible that there are additional permissions required for Chromecast discovery on Android devices. While you've added the INTERNET, ACCESS_NETWORK_STATE, and ACCESS_WIFI_STATE permissions, there might be other permissions specifically related to network discovery or multicast communication that are needed.

1. android.permission.CHANGE_WIFI_STATE
2. android.permission.ACCESS_FINE_LOCATION
3. android.permission.WAKE_LOCK

If you've tried all of these suggestions and you're still unable to discover Chromecast devices on Android, it might be helpful to report the issue to the Uno Platform community or the GitHub repository for the UnoChromeCast project. They might have encountered similar issues with Android devices and can provide more specific guidance.

**Reply by johnsmith3321** [reply]
You can download TV Cast Pro Application Free of cost from BlackAPKs.

[https://blackapks.com/en/tv-cast-pro](https://blackapks.com/en/tv-cast-pro)
